File: seckill-frontend/src/stores/seckill.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import request from '@/utils/request'

export const useSeckillStore = defineStore('seckill', () => {
  // 状态
  const products = ref<any[]>([])
  const currentStock = ref<{ product_1: number; product_2: number }>({ 
    product_1: 0, 
    product_2: 0 
  })
  const statistics = ref({
    success: 0,
    fail: 0,
    total: 0
  })
  const loading = ref(false)

  // 获取商品列表
  async function fetchProducts() {
    try {
      const res = await request.get('/product/list')
      products.value = res.data || []
    } catch (error) {
      console.error('获取商品列表失败:', error)
    }
  }

  // 获取实时库存和统计
  async function fetchStats() {
    try {
      const res = await request.get('/seckill/stats')
      if (res.data) {
        currentStock.value = res.data.inventory || { product_1: 0, product_2: 0 }
        statistics.value = res.data.statistics || { success: 0, fail: 0, total: 0 }
      }
    } catch (error) {
      console.error('获取统计数据失败:', error)
    }
  }

  // 执行秒杀
  async function doSeckill(userId: number, productId: number, quantity: number = 1) {
    loading.value = true
    try {
      const res = await request.post('/seckill/do', {
        userId,
        productId,
        quantity
      })
      return res.data
    } catch (error) {
      throw error
    } finally {
      loading.value = false
    }
  }

  // AI分析
  async function aiAnalyze(question: string) {
    try {
      const res = await request.post('/ai/analyze', {
        question,
        enable_reflection: true
      })
      return res.data
    } catch (error) {
      throw error
    }
  }

  return {
    products,
    currentStock,
    statistics,
    loading,
    fetchProducts,
    fetchStats,
    doSeckill,
    aiAnalyze
  }
})

File: python-ai-agent/app/api/seckill.py
"""
秒杀分析API - 被Java服务调用
"""
from fastapi import APIRouter
from pydantic import BaseModel
from typing import Dict, Any
from datetime import datetime
from app.services.redis_client import redis_client
from app.agent.brain import agent_brain

router = APIRouter(prefix="/api/seckill", tags=["秒杀分析"])

class QuestionRequest(BaseModel):
    question: str
    enable_reflection: bool = True

@router.post("/analyze")
async def analyze_seckill(request: QuestionRequest):
    """AI分析秒杀问题 - Java秒杀服务调用此接口"""
    await redis_client.connect()
    
    # 获取实时数据
    stock1 = await redis_client.get_seckill_stock(1)
    stock2 = await redis_client.get_seckill_stock(2)
    stats = await redis_client.get_seckill_stats()
    
    # 构建上下文
    context = f"""
    当前秒杀实时数据：
    - iPhone15库存: {stock1}
    - 华为Mate60库存: {stock2}
    - 秒杀成功数: {stats['success']}
    - 秒杀失败数: {stats['fail']}
    - 成功率: {(stats['success']/stats['total']*100) if stats['total']>0 else 0:.1f}%
    """
    
    # AI思考
    answer, confidence = agent_brain.think(request.question, context)
    
    # 低置信度时自我反思
    if request.enable_reflection and confidence < 0.7:
        answer = agent_brain.self_reflect(request.question, answer)
        confidence = min(confidence + 0.2, 0.95)
    
    return {
        "code": 200,
        "data": {
            "answer": answer,
            "confidence": confidence,
            "data_snapshot": {
                "stock": {"product_1": stock1, "product_2": stock2},
                "statistics": stats,
                "timestamp": datetime.now().isoformat()
            },
            "reflection_applied": confidence > 0.7 and request.enable_reflection
        }
    }

@router.get("/stats")
async def get_stats():
    """获取统计数据"""
    await redis_client.connect()
    stats = await redis_client.get_seckill_stats()
    stock1 = await redis_client.get_seckill_stock(1)
    stock2 = await redis_client.get_seckill_stock(2)
    return {"code": 200, "data": {"inventory": {"product_1": stock1, "product_2": stock2}, "statistics": stats}}